#VRML_SIM R2021b utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  randomSeed 16
}
Viewpoint {
  orientation 0.422 0.811 0.405 4.47
  position -3.13 4.25 -0.0192
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.5 2.5
  floorTileSize 5 5
}
WoodenBox {
  translation -0.375 0.05 0.375
  rotation 0 -1 0 0.3468
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.375 0.05 -0.375
  rotation 0 -1 0 0.34688418883387256
  name "wooden box(2)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.375 0.05 0.0
  rotation 0 -1 0 0.3468
  name "wooden box(3)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.375 0.05 0.0
  rotation 0 -1 0 0.3468
  name "wooden box(4)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 1.0 0.05 0.0
  rotation 0 -1 0 0.3468
  name "wooden box(5)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -1.0 0.05 0.0
  rotation 0 -1 0 0.3468
  name "wooden box(6)"
  size 0.1 0.1 0.1
  mass 0.2
}
Robot {
  translation 0.0 0 0
  rotation 0 0 0 0
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF SENSOR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.305882 0.305882
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
    GPS {
    }
    Compass {
      translation 0 0.05 0
    }
    Pen {
      translation 0 0.001 0
      inkColor 1 0 0
      inkDensity 1.0
      leadSize 0.005
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "robot"
  battery [
    250, 250, 30
  ]
  cpuConsumption 0.5
}
Charger {
  translation 0 0 0
  children [
    Transform {
      translation 0 0.15 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            transparency 0.20000000298023224
            roughness 0.2189099510000001
            metalness 0
            emissiveColor 1 0 0
          }
          geometry DEF CHARGER_GEOM Cylinder {
            height 0.3
            radius 0.1
            subdivision 24
          }
        }
        PointLight {
          attenuation 0 0 12
          color 1 0 0
          intensity 0.3
        }
      ]
    }
  ]
  name "charger0"
  battery [
    1000, 1000, 30
  ]
  radius 0.1
}

DEF GEO_FENCE_1 Solid {
  translation 0.0 0.001 1.0
  children [
    DEF GEO_FENCE_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.0 0.0
        roughness 0
        metalness 0
        transparency 0.6
      }
      geometry Box {
        size 0.5 0.00001 0.5
      }
    }
  ]
  name "solid(5)"
  boundingObject USE GEO_FENCE_SHAPE
}

DEF GEO_FENCE_2 Solid {
  translation 0.0 0.001 -1.0
  children [
    USE GEO_FENCE_SHAPE
  ]
  name "solid(6)"
  boundingObject USE GEO_FENCE_SHAPE
}
Robot {
  translation -0.75 0 0.75
  rotation 0 0 0 0
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1.0 0.0 0.0
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF SENSOR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.305882 0.305882
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "slave1"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "slave"
  cpuConsumption 0.5
}
Robot {
  translation 0.75 0 -0.75
  rotation 0 0 0 0
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1.0 0.0 0.0
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF SENSOR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.305882 0.305882
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "slave2"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "slave"
  cpuConsumption 0.5
}
Robot {
  translation -0.5 0 0.5
  rotation 0 0 0 0
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1.0 0.0 0.0
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF SENSOR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.305882 0.305882
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "slave3"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "slave"
  cpuConsumption 0.5
}
Robot {
  translation 0.5 0 -0.5
  rotation 0 0 0 0
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1.0 0.0 0.0
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 1 0 0 4.98467
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 1 0 0 4.58735
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF SENSOR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.305882 0.305882
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "slave4"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "slave"
  cpuConsumption 0.5
}

DEF WAYPOINT_1 Solid {
  translation 1.0 0.001 1.0
  children [
    DEF WAYPOINT_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.0 0.0
        roughness 0
        metalness 0
      }
      geometry Cylinder {
        radius 0.01,
        height 0.00001
      }
    }
  ]
  name "waypoint1"
}
DEF WAYPOINT_2 Solid {
  translation -1.0 0.001 1.0
  children [
    USE WAYPOINT_SHAPE
  ]
  name "waypoint2"
}
DEF WAYPOINT_3 Solid {
  translation -1.0 0.001 -1.0
  children [
    USE WAYPOINT_SHAPE
  ]
  name "waypoint3"
}
DEF WAYPOINT_4 Solid {
  translation 1.0 0.001 -1.0
  children [
    USE WAYPOINT_SHAPE
  ]
  name "waypoint4"
}